app/controllers/coronavirus/announcements_controller.rb

```diff
@@ -10,27 +10,34 @@ def new
 
     def create
       @announcement = page.announcements.new(announcement_params)
-      if @announcement.save && draft_updater.send
-        redirect_to coronavirus_page_path(page.slug), notice: helpers.t("coronavirus.announcements.create.success")
-      else
-        render :new
+
+      unless @announcement.valid?
+        render :new, status: :unprocessable_entity
+        return
       end
+
+      Announcement.transaction do
+        @announcement.save!
+        draft_updater.send
+      end
+
+      redirect_to coronavirus_page_path(page.slug), notice: helpers.t("coronavirus.announcements.create.success")
+    rescue Pages::DraftUpdater::DraftUpdaterError => e
+      flash[:alert] = e.message
+      render :new
```

Comment:
This should return an error status code too, there is precedent with using `status: :internal_server_error` from the subsections: https://github.com/alphagov/collections-publisher/blob/2c3acd32c13adfa6be393adadd8d3bc5ee8a89cd/app/controllers/coronavirus/sub_sections_controller.rb#L25

Reply:
I've updated the code to report a 5xx as suggested, though it's worth nothing that the error with publishing-api could be because some invalid content has been merged into GitHub, so whilst it's not an error create by this specific user it _is_ user error and could report an `unprocessable_entity` error.

Reply:
I see what you mean there but I don't think that fits the intent of a client error status code. My understanding of 4xx status codes is that you can resolve them by changing your request, in particular for a 422 it'd be changing the data you submit. Since GitHub data being wrong is outside of what can be achieved in this request that indicates to me that our server is the system with the error, and thus in the 5xx realm.